acceptance/tests/linker_variable_cleanup.rb

```diff
@@ -0,0 +1,115 @@
+require 'puppet/acceptance/common_utils'
+extend Puppet::Acceptance::CommandUtils
+
+confine :except, :platform => 'windows'
+
+def libsource(host)
+  kernel = on(host, facter("kernel")).stdout.chomp.downcase
+  arch = on(host, facter("architecture")).stdout.chomp.downcase
+
+  case kernel
+  when "linux"
+    case arch
+    when "x86_64", "amd64"
+      "libexplode.so.linux64"
+    when "i386"
+      "libexplode.so.linux32"
+    else
+      nil
+    end
+  when "aix"
+    "libexplode.so.aix"
+  when "sunos"
+    case arch
+    when "i86sol"
+      "libexplode.so.solaris"
+    else
+      nil
+    end
+  when "darwin"
+    "libexplode.dylib"
+  else
+    nil
+  end
+end
+
+def libdest(host)
+  kernel = on(host, facter("kernel")).stdout.chomp.downcase
+
+  case kernel
+  when "darwin"
+    "libobjc.A.dylib"
+  when "aix"
+    "libruby.so"
+  else
+    "libm.so.6"
+  end
+end
+
+test_name 'PA-437: cleanup linker environment' do
+  skip_test 'requires wrapper script which is created by the AIO' if [:gem, :git].include?(@options[:type])
+
+  fixtures = File.dirname(__FILE__) + "/../fixtures/dyld/"
+  dirs = {}
+  libs = {}
+
+  step "Lay down a crashing library on the SUT" do
+    agents.each do |agent|
+      source = libsource(agent)
+      if source == nil then
+        libs[agent] = nil
+        dirs[agent] = nil
+        next
+      end
+      dirs[agent] = agent.tmpdir("lib") + "/"
+      libs[agent] = dirs[agent] + libdest(agent)
+      fixture = fixtures + source
+
+      scp_to(agent, fixture, libs[agent])
+    end
+  end
+
+  teardown do
+    agents.each do |agent|
+      next if dirs[agent] == nil
+      on agent, "rm -rf #{dirs[agent]}"
+    end
+  end
+
+  step "Try to preload the crashing library" do
+    agents.each do |agent|
+      next if dirs[agent] == nil
+      kernel = on(agent, facter("kernel")).stdout.downcase.strip
+      var = case kernel
+            when "darwin"
+              "DYLD_INSERT_LIBRARIES"
+            when "aix"
+              "LDR_PRELOAD"
+            else
+              "LD_PRELOAD"
+            end
+      on agent, puppet("--version"), :environment => {var => libs[agent] }, :acceptable_exit_codes => [42]
+    end
+  end
+
+  step "Put the crashing library on the load path" do
+    agents.each do |agent|
+      next if dirs[agent] == nil
```

Comment:
Falling through all the way to success on an unsupported platform opens up the possibility that we might stop or not be testing the platforms we think we are. I think it would be safer if we explicitly target only the platforms we want to test, always run on those platforms, and fail down the road if for some reason we won't have the right fixtures/pathing etc.